src/auth/auth.service.ts

```diff
@@ -301,12 +305,37 @@ export class AuthService {
   }
 
   deleteUser = async (email: string) => {
-    const resp = await this.userRepo.delete({ email });
-    if (resp.affected == 0) {
+    try {
+      const user = await this.userRepo.findOne({
+        where: { email },
+      });
+
+      if (!user) {
+        throw new BadRequestException(
+          ZaLaResponse.BadRequest(
+            'User Not found',
+            'The Email Does not exists',
+            '400',
+          ),
+        );
+      }
+      //  send Delete request to the profile service to delete the user profile
+      // Axios
+      const coreUrl = `${this.configService.get<string>(
+        configConstant.baseUrls.coreService,
+      )}/profile/${user.profileID}`;
+      console.log(coreUrl);
+
+      await this.httpService.axiosRef({
+        method: 'delete',
+        url: coreUrl,
+      });
+
+      return await this.userRepo.delete({ email });
```

Comment:
You should delete the user first before making the http request. What if for some reasons the user wasn't deleted, and a request has been dispatched already.

Also remove the console log

Reply:
If the user is deleted first  I won't be able to get its profile ID

Reply:
I dont know if it's possible to return a deleted data

Reply:
It is possible if you use querybuilder, but why don't you assign the value to a variable before deleting, then use it afterwards

Reply:
Ok yes I think that will do